torchvision/datasets/imagenet.py

```diff
@@ -118,41 +114,97 @@ def extra_repr(self):
         return "Split: {split}".format(**self.__dict__)
 
 
-def parse_devkit(root):
-    idx_to_wnid, wnid_to_classes = parse_meta(root)
-    val_idcs = parse_val_groundtruth(root)
+def parse_devkit_archive(archive, meta_file=None):
+    """Parse the devkit archive of the ImageNet2012 classification dataset and save
```

Comment:
If they are not made static methods (see comment above) should we cross link to the class in the docstring? If yes, how do I need to write it?

Reply:
Do we want to expose those functions to the user, or is it something internal to the dataset? If it's internal, then I think it doesn't matter.

Basically, do you see an user ever calling this function?

Reply:
All these functions need to be called once before the dataset is ready for usage. They are called as part of the `download()` method, but since we keep this only for BC. Thus, every new user that doesn't use `download=True` will have to call these functions separately.